nilearn/_utils/testing.py

```diff
@@ -161,11 +161,11 @@ def write_tmp_imgs(*imgs, **kwargs):
             with warnings.catch_warnings():
                 warnings.simplefilter("ignore", RuntimeWarning)
                 for img in imgs:
-                    filename = tempfile.mktemp(prefix=prefix,
-                                               suffix=suffix,
-                                               dir=None)
-                    filenames.append(filename)
-                    img.to_filename(filename)
+                    fd, path = tempfile.mkstemp(prefix=prefix,
```

Comment:
you need to `os.close(fd)`

Reply:
btw this function has another issue: if removing one image fails (for example if it has been removed by the caller) the other ones are not removed